app.py

```diff
@@ -20,16 +20,35 @@ def home() -> str:
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
     animals = Animal.query.all()
+
     return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
 
 
+@app.route('/search', methods=['GET', 'POST'])
+def search() -> Response:
+    filter_param = request.args.get('FilterInput')
+    if filter_param:
+        animals = Animal.query.filter_by(name=filter_param).all()
```

Comment:
You can filter even by substring (part of name)
`Model.query.filter(Model.columnName.contains('sub_string'))`